<Type Name="EntityDataSourceValidationException" FullName="System.Web.UI.WebControls.EntityDataSourceValidationException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="bcf808c55758b84c73449fe81aed7c92f5cb5fda" /><Meta Name="ms.sourcegitcommit" Value="c982ad9e10a242d90f1e631b5981570f9e5d9f34" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="zh-CN" /><Meta Name="ms.lasthandoff" Value="09/10/2020" /><Meta Name="ms.locfileid" Value="89680327" /></Metadata><TypeSignature Language="C#" Value="public sealed class EntityDataSourceValidationException : Exception, System.Web.DynamicData.IDynamicValidatorException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable sealed beforefieldinit EntityDataSourceValidationException extends System.Exception implements class System.Web.DynamicData.IDynamicValidatorException" />
  <TypeSignature Language="DocId" Value="T:System.Web.UI.WebControls.EntityDataSourceValidationException" />
  <TypeSignature Language="VB.NET" Value="Public NotInheritable Class EntityDataSourceValidationException&#xA;Inherits Exception&#xA;Implements IDynamicValidatorException" />
  <TypeSignature Language="C++ CLI" Value="public ref class EntityDataSourceValidationException sealed : Exception, System::Web::DynamicData::IDynamicValidatorException" />
  <TypeSignature Language="F#" Value="type EntityDataSourceValidationException = class&#xA;    inherit Exception&#xA;    interface IDynamicValidatorException" />
  <AssemblyInfo>
    <AssemblyName>System.Web.Entity</AssemblyName>
    <AssemblyVersion>3.5.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
  </Base>
  <Interfaces>
    <Interface>
      <InterfaceName>System.Web.DynamicData.IDynamicValidatorException</InterfaceName>
    </Interface>
  </Interfaces>
  <Attributes>
    <Attribute>
      <AttributeName Language="C#">[System.Serializable]</AttributeName>
      <AttributeName Language="F#">[&lt;System.Serializable&gt;]</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary><span data-ttu-id="668aa-101">表示验证动态数据源的属性时发生的错误。</span><span class="sxs-lookup"><span data-stu-id="668aa-101">Represents errors that occur when validating properties of a dynamic data source.</span></span></summary>
    <remarks>To be added.</remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Web.Entity</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="668aa-102">创建 <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="668aa-102">Creates a new instance of the <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public EntityDataSourceValidationException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.EntityDataSourceValidationException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; EntityDataSourceValidationException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Entity</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="668aa-103">使用系统提供用以描述错误的消息创建 <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="668aa-103">Creates a new instance of the <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> class with a system-supplied message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="668aa-104">此构造函数将新实例的 <xref:System.Exception.Message%2A> 属性初始化为系统提供的消息，该消息描述诸如“DefaultMessageDisplayedByParameterlessCtorWriterMustSupply”的错误。</span><span class="sxs-lookup"><span data-stu-id="668aa-104">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "DefaultMessageDisplayedByParameterlessCtorWriterMustSupply".</span></span> <span data-ttu-id="668aa-105">此消息会考虑当前系统区域性。</span><span class="sxs-lookup"><span data-stu-id="668aa-105">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="668aa-106">下表显示了 <xref:System.InsufficientMemoryException> 实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="668aa-106">The following table shows the initial property values for an instance of <xref:System.InsufficientMemoryException>.</span></span>  
  
|<span data-ttu-id="668aa-107">properties</span><span class="sxs-lookup"><span data-stu-id="668aa-107">Property</span></span>|<span data-ttu-id="668aa-108">值</span><span class="sxs-lookup"><span data-stu-id="668aa-108">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="668aa-109">`null`.</span><span class="sxs-lookup"><span data-stu-id="668aa-109">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="668aa-110">本地化的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="668aa-110">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="/dotnet/standard/exceptions/"><span data-ttu-id="668aa-111">处理和引发异常</span><span class="sxs-lookup"><span data-stu-id="668aa-111">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public EntityDataSourceValidationException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.EntityDataSourceValidationException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; EntityDataSourceValidationException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.EntityDataSourceValidationException : string -&gt; System.Web.UI.WebControls.EntityDataSourceValidationException" Usage="new System.Web.UI.WebControls.EntityDataSourceValidationException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Entity</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="668aa-112">描述该异常的消息。</span><span class="sxs-lookup"><span data-stu-id="668aa-112">The message that describes the exception.</span></span> <span data-ttu-id="668aa-113">此构造函数的调用方需要确保此字符串已针对当前系统区域性进行了本地化。</span><span class="sxs-lookup"><span data-stu-id="668aa-113">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span></param>
        <summary><span data-ttu-id="668aa-114">使用指定的错误描述消息创建 <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="668aa-114">Creates a new instance of the <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> class with a specified message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="668aa-115">下表显示了 <xref:System.Web.UI.WebControls.EntityDataSourceValidationException> 实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="668aa-115">The following table shows the initial property values for an instance of <xref:System.Web.UI.WebControls.EntityDataSourceValidationException>.</span></span>  
  
|<span data-ttu-id="668aa-116">properties</span><span class="sxs-lookup"><span data-stu-id="668aa-116">Property</span></span>|<span data-ttu-id="668aa-117">值</span><span class="sxs-lookup"><span data-stu-id="668aa-117">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="668aa-118">`null`.</span><span class="sxs-lookup"><span data-stu-id="668aa-118">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="668aa-119">`message` 中指定的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="668aa-119">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="/dotnet/standard/exceptions/"><span data-ttu-id="668aa-120">处理和引发异常</span><span class="sxs-lookup"><span data-stu-id="668aa-120">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public EntityDataSourceValidationException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.EntityDataSourceValidationException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; EntityDataSourceValidationException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Web.UI.WebControls.EntityDataSourceValidationException : string * Exception -&gt; System.Web.UI.WebControls.EntityDataSourceValidationException" Usage="new System.Web.UI.WebControls.EntityDataSourceValidationException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Entity</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="668aa-121">描述该异常的消息。</span><span class="sxs-lookup"><span data-stu-id="668aa-121">The message that describes the exception.</span></span> <span data-ttu-id="668aa-122">此构造函数的调用方需要确保此字符串已针对当前系统区域性进行了本地化。</span><span class="sxs-lookup"><span data-stu-id="668aa-122">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span></param>
        <param name="innerException"><span data-ttu-id="668aa-123">导致当前异常的异常。</span><span class="sxs-lookup"><span data-stu-id="668aa-123">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="668aa-124">如果 <paramref name="innerException" /> 参数不为 <see langword="null" />，则当前异常将在处理内部异常的 <see langword="catch" /> 块中引发。</span><span class="sxs-lookup"><span data-stu-id="668aa-124">If the <paramref name="innerException" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="668aa-125">使用指定的错误消息和对导致此异常的内部异常的引用创建 <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="668aa-125">Creates a new instance of the <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="668aa-126">因前一个异常而直接引发的异常应在 <xref:System.Exception.InnerException%2A> 属性中包含对前一个异常的引用。</span><span class="sxs-lookup"><span data-stu-id="668aa-126">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="668aa-127"><xref:System.Exception.InnerException%2A> 属性将返回传递给构造函数的同一值；如果 <xref:System.Exception.InnerException%2A> 属性不向构造函数提供内部异常值，则返回 `null`。</span><span class="sxs-lookup"><span data-stu-id="668aa-127">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="668aa-128">下表显示了 <xref:System.Web.UI.WebControls.EntityDataSourceValidationException> 实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="668aa-128">The following table shows the initial property values for an instance of <xref:System.Web.UI.WebControls.EntityDataSourceValidationException>.</span></span>  
  
|<span data-ttu-id="668aa-129">properties</span><span class="sxs-lookup"><span data-stu-id="668aa-129">Property</span></span>|<span data-ttu-id="668aa-130">值</span><span class="sxs-lookup"><span data-stu-id="668aa-130">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="668aa-131">`null`.</span><span class="sxs-lookup"><span data-stu-id="668aa-131">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="668aa-132">`message` 中指定的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="668aa-132">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="/dotnet/standard/exceptions/"><span data-ttu-id="668aa-133">处理和引发异常</span><span class="sxs-lookup"><span data-stu-id="668aa-133">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName="GetObjectData">
      <MemberSignature Language="C#" Value="public override void GetObjectData (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig virtual instance void GetObjectData(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Web.UI.WebControls.EntityDataSourceValidationException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Public Overrides Sub GetObjectData (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; override void GetObjectData(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberSignature Language="F#" Value="override this.GetObjectData : System.Runtime.Serialization.SerializationInfo * System.Runtime.Serialization.StreamingContext -&gt; unit" Usage="entityDataSourceValidationException.GetObjectData (info, context)" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Web.Entity</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Void</ReturnType>
      </ReturnValue>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" Index="0" FrameworkAlternate="netframework-3.5;netframework-4.0" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" Index="1" FrameworkAlternate="netframework-3.5;netframework-4.0" />
      </Parameters>
      <Docs>
        <param name="info"><span data-ttu-id="668aa-134">承载序列化对象数据的对象。</span><span class="sxs-lookup"><span data-stu-id="668aa-134">The object that holds the serialized object data.</span></span></param>
        <param name="context"><span data-ttu-id="668aa-135">关于来源和目标的上下文信息</span><span class="sxs-lookup"><span data-stu-id="668aa-135">The contextual information about the source or destination.</span></span></param>
        <summary><span data-ttu-id="668aa-136">使用序列化的数据创建 <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="668aa-136">Creates a new instance of the <see cref="T:System.Web.UI.WebControls.EntityDataSourceValidationException" /> class with serialized data.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  

<span data-ttu-id="668aa-137">在反序列化期间调用此构造函数以重新构造在流上传输的异常对象。</span><span class="sxs-lookup"><span data-stu-id="668aa-137">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span>
<span data-ttu-id="668aa-138">有关详细信息，请参阅 [XML 和 SOAP 序列化](/dotnet/standard/serialization/xml-and-soap-serialization)。</span><span class="sxs-lookup"><span data-stu-id="668aa-138">For more information, see [XML and SOAP Serialization](/dotnet/standard/serialization/xml-and-soap-serialization).</span></span>

 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="System.Web.DynamicData.IDynamicValidatorException.InnerExceptions">
      <MemberSignature Language="C#" Value="System.Collections.Generic.IDictionary&lt;string,Exception&gt; System.Web.DynamicData.IDynamicValidatorException.InnerExceptions { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance class System.Collections.Generic.IDictionary`2&lt;string, class System.Exception&gt; System.Web.DynamicData.IDynamicValidatorException.InnerExceptions" />
      <MemberSignature Language="DocId" Value="P:System.Web.UI.WebControls.EntityDataSourceValidationException.System#Web#DynamicData#IDynamicValidatorException#InnerExceptions" />
      <MemberSignature Language="VB.NET" Value=" ReadOnly Property InnerExceptions As IDictionary(Of String, Exception) Implements IDynamicValidatorException.InnerExceptions" />
      <MemberSignature Language="C++ CLI" Value="property System::Collections::Generic::IDictionary&lt;System::String ^, Exception ^&gt; ^ System::Web::DynamicData::IDynamicValidatorException::InnerExceptions { System::Collections::Generic::IDictionary&lt;System::String ^, Exception ^&gt; ^ get(); };" />
      <MemberSignature Language="F#" Value="member this.System.Web.DynamicData.IDynamicValidatorException.InnerExceptions : System.Collections.Generic.IDictionary&lt;string, Exception&gt;" Usage="System.Web.DynamicData.IDynamicValidatorException.InnerExceptions" />
      <MemberType>Property</MemberType>
      <Implements>
        <InterfaceMember>P:System.Web.DynamicData.IDynamicValidatorException.InnerExceptions</InterfaceMember>
      </Implements>
      <AssemblyInfo>
        <AssemblyName>System.Web.Entity</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Collections.Generic.IDictionary&lt;System.String,System.Exception&gt;</ReturnType>
      </ReturnValue>
      <Docs>
        <summary><span data-ttu-id="668aa-139">获取在对新数据字段或编辑的数据字段进行验证时发生的异常。</span><span class="sxs-lookup"><span data-stu-id="668aa-139">Gets the exceptions that occur when a new or edited data field is validated.</span></span></summary>
        <value><span data-ttu-id="668aa-140">一个集合，其中包含发生的所有异常。</span><span class="sxs-lookup"><span data-stu-id="668aa-140">A collection that contains all the exceptions that occurred.</span></span></value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="668aa-141">此成员是显式接口成员的实现。</span><span class="sxs-lookup"><span data-stu-id="668aa-141">This member is an explicit interface member implementation.</span></span> <span data-ttu-id="668aa-142">它只能在 <xref:System.Web.UI.WebControls.EntityDataSourceValidationException> 实例被强制转换为 <xref:System.Web.DynamicData.IDynamicValidatorException> 接口时使用。</span><span class="sxs-lookup"><span data-stu-id="668aa-142">It can be used only when the <xref:System.Web.UI.WebControls.EntityDataSourceValidationException> instance is cast to an <xref:System.Web.DynamicData.IDynamicValidatorException> interface.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>
